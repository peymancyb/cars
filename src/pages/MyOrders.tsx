import React, {useState} from 'react';
import LocalStorage from '../api/localStorage';
import CarList from '../components/CarList';
import Layout from '../components/Layout';

function MyOrders() {
  const [cars, setCars] = useState(LocalStorage.getCars());

  const onRemoveCar = (stockNumber: number) => {
    LocalStorage.removeCar(stockNumber);
    const newCarList = LocalStorage.getCars();
    setCars(newCarList);
  };

  return (
    <Layout>
      {!cars.length ? (
        <p>Empty list!</p>
      ) : (
        <CarList
          loading={false}
          cars={cars}
          removable
          onRemoveItem={onRemoveCar}
        />
      )}
    </Layout>
  );
}
export default MyOrders;
