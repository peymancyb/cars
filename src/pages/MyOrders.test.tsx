import React from 'react';
import {render, cleanup} from '@testing-library/react';
import MyOrders from './MyOrders';
import LocalStorage from '../api/localStorage';

const carMockData = {
  stockNumber: Math.random() * 100 + 1,
  manufacturerName: '',
  modelName: '',
  color: '',
  mileage: {
    number: 1,
    unit: 'km',
  },
  fuelType: '',
  pictureUrl: '',
};

afterEach(cleanup);

test('(MyOrders) component should match snapshot', () => {
  const {asFragment} = render(<MyOrders />);
  expect(asFragment).toMatchSnapshot();
});

test('(MyOrders) component renders correctly', () => {
  const {getByTestId} = render(<MyOrders />);
  expect(getByTestId('my-orders-component')).toBeTruthy();
});

test('(MyOrders) should show empty message when there is no order', () => {
  const {getByText} = render(<MyOrders />);

  expect(getByText('Empty list!')).toBeTruthy();
});

test('(MyOrders) should render car list using <CarList /> while there are car saved in localStorage', () => {
  LocalStorage.addCar(carMockData);
  const {getByTestId} = render(<MyOrders />);

  expect(getByTestId('car-list-component')).toBeTruthy();
  localStorage.clear();
});
