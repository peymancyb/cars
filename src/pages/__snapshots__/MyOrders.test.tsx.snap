// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(MyOrders) component should match snapshot 1`] = `[Function]`;
